src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ListMutatorFactory.java

```diff
@@ -97,18 +112,74 @@ public void initInPlace(List<T> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(List<T> list, PseudoRandom prng, int minSize) {
+      int chunkSize =
+          prng.closedRange(1, Math.max(1, Math.min(list.size() - minSize, list.size() / 2)));
+      int chunkOffset = prng.closedRange(0, list.size() - chunkSize);
+      Iterator<T> iterator = list.listIterator(chunkOffset);
+      int elementsRemoved = 0;
+      while (iterator.hasNext() && elementsRemoved < chunkSize) {
+        iterator.next();
+        iterator.remove();
+        elementsRemoved++;
+      }
+    }
+
+    private void insertRandomChunk(List<T> list, PseudoRandom prng, int maxSize) {
+      // Enforces slower growth for smaller lists and lists that are close to
+      // `maxSize`.
+      int chunkSize = prng.closedRange(1, Math.min(maxSize - list.size(), list.size()));
+      int chunkOffset = prng.indexIn(list);
+      for (int i = 0; i < chunkSize; i++) {
+        T newElement = elementMutator.init(prng);
+        list.add(chunkOffset + i, newElement);
```

Comment:
This presumably requires quite a bit of copying - I expect this to be on the order of <chunk size> * <list size> as it needs to copy over the tail of the list in each iteration (see https://github.com/openjdk/jdk/blob/9df20600592427550998c6685f103737e3115a51/src/java.base/share/classes/java/util/ArrayList.java#L487-L489).

Instead, you could do a batch `addAll` at the correct position. This requires collecting the new elements in temporary new list (which you could initialize with the correct capacity `chunkSize` directly).